app/models/miq_filter.rb

```diff
@@ -1,17 +1,48 @@
 module MiqFilter
+  ALLOWED_DESCENDANT_CLASSES_FROM_MODEL = %w[ExtManagementSystem].freeze
+
   def self.belongsto2object(tag)
     belongsto2object_list(tag).last
   end
 
+  def self.find_descendant_class_by(klass, name)
+    if ALLOWED_DESCENDANT_CLASSES_FROM_MODEL.include?(klass.to_s) && name.end_with?(ManageIQ::Providers::NetworkManager::PROVIDER_NAME)
```

Comment:
@libor Perhaps this logic can be moved into the `NetworkManager` class and this code can be something like
```ruby
if ALLOWED_DESCENDANT_CLASSES_FROM_MODEL.include?(klass.to_s) && klass.respond_to?(:belongsto_descendant_class)
 return  klass.belongsto_descendant_class(name)
else
 ...
```

The in `NetworkManager`
```ruby
def belongsto_descendant_class(name)
  ManageIQ::Providers::NetworkManager if name.end_with?(ManageIQ::Providers::NetworkManager::PROVIDER_NAME)
end
```
This way the provider specific code doesn't have to live in `Filterer`

Reply:
👍, I followed pattern but I need to put to the `ExtManagementSystem` because klass is `ExtManagementSystem` , thanks!